external_search.py

```diff
@@ -268,375 +279,59 @@ def base_index_name(self, index_or_alias):
-        if audiences:
-            if isinstance(audiences, list) or isinstance(audiences, set):
-                audiences = [_f(aud) for aud in audiences]
-                clauses.append(dict(terms=dict(audience=audiences)))
-        if target_age:
-            if isinstance(target_age, tuple) and len(target_age) == 2:
-                lower, upper = target_age
-            else:
-                lower = target_age.lower
-                upper = target_age.upper
-
-            age_clause = {
-                "and": [
-                    {
-                        "or" : [
-                            {"range": {"target_age.upper": {"gte": lower}}},
-                            {
-                                "bool": {
-                                    "must_not" : {
-                                        "exists": {"field" : "target_age.upper"}
-                                    }
-                                }
-                            }
-                        ]
-                    },
-                    {
-                        "or" : [
-                            {"range": {"target_age.lower": {"lte": upper}}},
-                            {
-                                "bool": {
-                                    "must_not" : {
-                                        "exists": {"field" : "target_age.lower"}
-                                    }
-                                }
-                            }
-                        ]
-                    }
-                ]
-            }
-            clauses.append(age_clause)
-        if len(clauses) > 0:
-            return {'and': clauses}
+        fields = None
+        if debug:
+            # Get some additional fields to make it easy to check whether
+            # we got reasonable looking results.
+            fields = ["_id", "title", "author", "license_pool_id"]
         else:
-            return {}
+            # All we absolutely need is the document ID, which is a
+            # key into the database.
+            fields = ["_id"]
+
+        # Change the Search object so it only retrieves the fields
+        # we're asking for.
+        search = search.fields(fields)
+
+        if not pagination:
+            from lane import Pagination
+            pagination = Pagination.default()
+        start = pagination.offset
+        stop = start + pagination.size
+
+        a = time.time()
+        # NOTE: This is the code that actually executes the ElasticSearch
+        # request.
+        results = search[start:stop]
+        if debug:
+            b = time.time()
+            self.log.info("Elasticsearch query completed in %.2fsec", b-a)
+            for i, result in enumerate(results):
+                self.log.info(
+                    '%02d "%s" (%s) work=%s score=%.3f',
+                    i, result.title, result.author, result.meta['id'],
+                    result.meta['score']
+                )
```

Comment:
You could try `search.extra(explain=True)` for debug mode - this adds extra info about how the score for each item was computed and it's very useful for debugging. It's much too verbose to log though.